R/parse_data_for_modelfit.R

```diff
@@ -0,0 +1,401 @@
+      -tidyselect::starts_with("AVISIT"), -tidyselect::starts_with("PARAM"),
+      -tidyselect::ends_with("TMF"), -tidyselect::starts_with("TRT"), -tidyselect::starts_with("ATPT"), -DRUG
+    )
+  
+  #---- Derive Covariates ----
+  # Include numeric values for STUDYIDN, USUBJIDN, SEXN, RACEN etc.
+  covar <- data$adsl %>%
+    dplyr::mutate(
+      STUDYIDN = as.numeric(stringr::word(USUBJID, 1, sep = stringr::fixed("-"))),
+      SITEIDN = as.numeric(stringr::word(USUBJID, 2, sep = stringr::fixed("-"))),
+      USUBJIDN = as.numeric(stringr::word(USUBJID, 3, sep = stringr::fixed("-"))),
+      SUBJIDN = as.numeric(SUBJID),
+      SEXN = dplyr::case_when(
+        SEX == "M" ~ 1,
+        SEX == "F" ~ 2,
+        TRUE ~ 3
+      ),
+      ROUTE = "oral",    # TODO: hardcoded for now
+      ROUTEN = 1,        # .
+      FORM = "tablet",   # .
+      FORMN = 1,         # /TODO
+    )
+  categorical_vars <- c("RACE", "ETHNIC", "ARM", "ACTARM", "COUNTRY")
+  covar[paste0(categorical_vars, "N")] <- data.matrix(covar[, categorical_vars])
+  covar <- covar %>%
+    dplyr::mutate(
+      COHORT = ARMN,
+      COHORTC = ARM
+    ) %>%
+    dplyr::select(
+      STUDYID, STUDYIDN, SITEID, SITEIDN, USUBJID, USUBJIDN,
+      SUBJID, SUBJIDN, AGE, SEX, SEXN, COHORT, COHORTC, ROUTE, ROUTEN,
+      RACE, RACEN, ETHNIC, ETHNICN, FORM, FORMN, COUNTRY, COUNTRYN
+    )
+  
+  #---- Derive additional baselines from VS and LB ----
+  numeric_vars <- c("CREAT", "ALT", "AST", "BILI") ## TODO: fairly generic, but might not always be available or might include others
+  labsbl <- data$lb %>%
+    dplyr::filter(LBBLFL == "Y" & LBTESTCD %in% numeric_vars) %>%
+    dplyr::mutate(LBTESTCDB = paste0(LBTESTCD, "BL")) %>%
+    dplyr::select(STUDYID, USUBJID, LBTESTCDB, LBSTRESN)
+  
+  covar_vslb <- covar %>%
+    admiral::derive_vars_merged(
+      dataset_add = data$vs,
+      filter_add = VSTESTCD == "HEIGHT",
+      by_vars = admiral::exprs(STUDYID, USUBJID),
+      new_vars = admiral::exprs(HTBL = VSSTRESN)
+    ) %>%
+    admiral::derive_vars_merged(
+      dataset_add = data$vs,
+      filter_add = VSTESTCD == "WEIGHT" & VSBLFL == "Y",
+      by_vars = admiral::exprs(STUDYID, USUBJID),
+      new_vars = admiral::exprs(WTBL = VSSTRESN)
+    ) %>%
+    admiral::derive_vars_transposed(
+      dataset_merge = labsbl,
+      by_vars = admiral::exprs(STUDYID, USUBJID),
+      key_var = LBTESTCDB,
+      value_var = LBSTRESN
+    ) %>%
+    dplyr::mutate(
+      BMIBL = admiral::compute_bmi(height = HTBL, weight = WTBL),
+      BSABL = admiral::compute_bsa(
+        height = HTBL,
+        weight = HTBL,
+        method = "Mosteller"
+      ),
+      CRCLBL = admiral::compute_egfr(
+        creat = CREATBL, creatu = "SI", age = AGE, weight = WTBL, sex = SEX,
+        method = "CRCL"
+      ),
+      EGFRBL = admiral::compute_egfr(
+        creat = CREATBL, creatu = "SI", age = AGE, wt = WTBL, sex = SEX,
```

Comment:
```suggestion
        creat = CREATBL, creatu = "SI", age = AGE, weight = WTBL, sex = SEX,
```
I missed another instance of this deprecated argument in my first review